docs/scripts/generateProptypes.ts

```diff
@@ -68,14 +68,30 @@ async function generateProptypes(program: ttp.ts.Program, sourceFile: string) {
       if (['children', 'state'].includes(prop.name) && component.name.startsWith('DataGrid')) {
         return false;
       }
-      let shouldDocument = true;
+      let shouldExclude = false;
       prop.filenames.forEach((filename) => {
-        // Don't include props from external dependencies
-        if (/node_modules/.test(filename)) {
-          shouldDocument = false;
```

Comment:
Maybe we should differentiate between peer dependencies and dependencies. For example, MUI X has the TextField as a peer dependency, so it doesn't control which props from the API are supported or not. So the generation could be off (tell you a prop value isn't there when using a version of MUI X that was built with mui/material@v5.6.0 while the user is on mui/material@v5.7.0.

Reply:
That is a good point I did not think about. 🤔 
So, all in all, MUI (core) preference is declare inheritance and don't focus on having correct/full PropTypes?

Reply:
If the prop-type is fully declared, then when the component receives a wrong prop, the component warns twice: once in the parent and once in the actually inherited component. It might be clearer to warn once and in the lower level. It teaches developers the lower level component structure.

Reply:
> MUI Core avoids duplicating the prop-types. If the prop-type is fully declared, then when the component receives a wrong prop, the component warns twice: once in the parent and once in the actually inherited component. It assumes it's clearer to not duplicate the prop-type, to warn once and in the lower level. It teaches developers the lower-level component structure.

It seems that this problem exists in MUI Core as well.
Did a small test with `DateField` component in a fresh JS project and my change added one extra warn/error as you mentioned, but there already are 4 others being thrown by the inherited components. 🙈 
![Screenshot 2022-12-30 at 11 21 09](https://user-images.githubusercontent.com/4941090/210054860-808e50dc-336f-465c-823c-b61d25ad90ac.png)

Reply:
The number 2 is strange on these component names.

Yeah, definitely. From what I recall, we decided to duplicate some of the props because the UX of the inherited prop is not good enough on the API pages. The relevant issue: https://github.com/mui/material-ui/issues/18288.